Decred Wallet/Features/Receive/ReceiveViewController.swift

```diff
@@ -0,0 +1,179 @@
+    @IBOutlet weak var selectedAccountView: UIView!
+    @IBOutlet weak var accountNameLabel: UILabel!
+    @IBOutlet weak var walletNameLabel: UILabel!
+    @IBOutlet weak var totalAccountBalanceLabel: UILabel!
+    @IBOutlet weak var addressQRCodeImageView: UIImageView!
+    @IBOutlet weak var walletAddressLabel: UILabel!
+    @IBOutlet weak var syncInProgressLabel: UILabel!
+
+    var tapGesture = UITapGestureRecognizer()
+    var oldAddress = ""
+
+    var selectedWallet: DcrlibwalletWallet?
+    var selectedAccount: DcrlibwalletAccount?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.displayAddressForFirstWalletAccount()
+    }
+
+    func setupUI() {
+        self.addressQRCodeImageView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.walletAddressLabel.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.copyAddress)))
+        self.selectedAccountView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.showAccountSelectorDialog)))
+    }
+
+    private func displayAddressForFirstWalletAccount() {
+        let accountsFilterFn: (DcrlibwalletAccount) -> Bool = { $0.totalBalance > 0 || $0.name != "imported" }
+        guard let wallet = WalletLoader.shared.wallets.map({ Wallet.init($0, accountsFilterFn: accountsFilterFn) }).first,
+            let account = wallet.accounts.first else { return }
+
+        self.updateSelectedAccount(wallet.id, account)
+    }
+
+    @objc func copyAddress() {
+        DispatchQueue.main.async {
+            UIPasteboard.general.string = self.walletAddressLabel.text!
+            Utils.showBanner(in: self.view, type: .success, text: LocalizedStrings.walletAddrCopied)
+        }
+    }
+
+    @objc func showAccountSelectorDialog(_ sender: Any) {
+        AccountSelectorDialog.show(sender: self,
+                                 title: LocalizedStrings.receivingAccount,
+                                 selectedWallet: selectedWallet,
+                                 selectedAccount: self.selectedAccount,
+                                 callback: self.updateSelectedAccount)
+    }
+
+    private func displayAddressAndQRCode(receiveAddress: String) {
+        DispatchQueue.main.async {
+            self.walletAddressLabel.text = receiveAddress
+            self.displayQRCodeImage(for: receiveAddress)
+        }
+    }
+
+    private func generateNewAddress() {
+        self.oldAddress = self.walletAddressLabel.text!
+        self.getNextAddress()
+    }
+
+    @objc private func getNextAddress() {
+        if let wallet = self.selectedWallet,
+            let account = self.selectedAccount {
+                let receiveAddress = wallet.nextAddress(account.number, error: nil)
+                if self.oldAddress != receiveAddress {
```

Comment:
No need for a `self.oldAddress` variable.
```suggestion
                if self.walletAddressLabel.text! != receiveAddress {
```